File: Chess.Angular/ClientApp/src/app/chess/models/chessUnit.ts
export class ChessUnit{
    
    readonly Type: UnitType;
    readonly Color: UnitColor;

    constructor(color: UnitColor, type: UnitType){
        this.Color = color;
        this.Type = type;
    }
}

export enum UnitType {
    Pawn = 0,
    Rook = 1,
    Knight = 2,
    Bishop = 3,
    Queen = 4,
    King = 5,
  }

export enum UnitColor {
    White = 0,
    Black = 1,
  }

File: Chess.Domain/Models/ChessUnit.cs
using Chess.Domain.Enums;

namespace Chess.Domain.Models
{
    public class ChessUnit
    {
        public ChessUnit(UnitType type, UnitColor color)
        {
            Type = type;
            Color = color;
        }

        public UnitType Type { get; }

        public UnitColor Color { get; }

    }
}
